File: Counting Valleys/counting-valleys.php
<?php
/**
 * Created by PhpStorm.
 * User: kovalevcon
 * Date: 07.09.18
 * Time: 18:04
 */

/**
 * Count valleys function
 *
 * @param int $n
 * @param string $s
 * @return int
 */
function countingValleys(int $n, string $s): int
{
    $count = $seaLevel = 0;
    for ($i = 0; $i < $n; $i++) {
        if ($s[$i] === 'U') {
            ++$seaLevel;
            if ($seaLevel === 0) {
                ++$count;
            }
        } else {
            --$seaLevel;
        }
    }

    return $count;
}

echo "<pre>countingValleys(8, 'UDDDUDUU')\r\n";
echo countingValleys(8, 'UDDDUDUU'); // 1
echo "</pre>";

echo "<pre>countingValleys(12, 'DDUUDDUDUUUD')\r\n";
echo countingValleys(12, 'DDUUDDUDUUUD'); // 2
echo "</pre>";

File: Counting Valleys/counting-valleys.py
def countingValleys(n, s):
    count, sea_level = [0, 0]
    for chr in list(s):
        if chr == 'U':
            sea_level += 1
            if sea_level == 0:
                count += 1
        else:
            sea_level -= 1

    return count


print(countingValleys(8, 'UDDDUDUU'))  # 1
print(countingValleys(12, 'DDUUDDUDUUUD'))  # 2